azurerm/internal/services/kusto/resource_arm_kusto_eventgrid_data_connection.go

```diff
@@ -0,0 +1,284 @@
+)
+
+func resourceArmKustoEventGridDataConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKustoEventGridDataConnectionCreateUpdate,
+		Read:   resourceArmKustoEventGridDataConnectionRead,
+		Update: resourceArmKustoEventGridDataConnectionCreateUpdate,
+		Delete: resourceArmKustoEventGridDataConnectionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoDataConnectionName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"cluster_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoClusterName,
+			},
+
+			"database_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoDatabaseName,
+			},
+
+			"storage_account_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"eventhub_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"consumer_group": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubConsumerName(),
+			},
+
+			"table_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMKustoEntityName,
+			},
+
+			"mapping_rule_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMKustoEntityName,
+			},
+
+			"data_format": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(kusto.MULTIJSON),
+					string(kusto.JSON),
+					string(kusto.CSV),
+					string(kusto.TSV),
+					string(kusto.SCSV),
+					string(kusto.SOHSV),
+					string(kusto.PSV),
+					string(kusto.TXT),
+					string(kusto.RAW),
+					string(kusto.SINGLEJSON),
+					string(kusto.AVRO),
+					string(kusto.TSVE),
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceArmKustoEventGridDataConnectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Kusto.DataConnectionsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure Kusto EventGrid Data Connection creation.")
```

Comment:
Data Explorer

Reply:
I'm not sure where Data Explorer comes from but this resource is called EventGridDataConnection and it'd be good to keep that matched in the logfile to prevent any confusion should anyone need to peruse the logfile for any reason.

Reply:
I meant: "[INFO] preparing arguments for Azure Data Explorer EventGrid Data Connection creation."

Reply:
`Azure Data Explorer` is just a marketing name. `Kusto` is the MS-internal name of the product, which will not change in the future. Therefore, the azure-sdk-for-go also uses the identifier `kusto` instead of `adx`or `dataexplorer`.